src/twisted/newsfragments/10231.bugfix

```diff
@@ -0,0 +1 @@
+Fix type annotations for inlineCallbacks.
```

Comment:
```suggestion
Improve type annotations for inlineCallbacks.
```

Reply:
this is an improvement, but it's not fixed: It's not fixable without fixing `Deferred.__iter__` and using ParamSpec to propagate the decorated call signatute

Reply:
If I understood this correctly.

```suggestion
twisted.internet.defer.inlineCallbacks has now improved type annotation that will not raise an error for non-object return values.
```


but basically this is like  no typing at all as it can return anything.

You can ignore my comment and I am still trying to figure it out why we need 100% typing in Python

Reply:
@adiroiban this isn't correct either. There's no such thing as "non-object return values"

Reply:
here's the issue and the fix more clearly: https://mypy-play.net/?mypy=latest&python=3.10&flags=strict&gist=64b950daeb501e6a6bd0592d9fe456d5

Reply:
I've updated the newsfile - hopefully it is better now?